### ISSUE TITLE
pytorch1.5 does not support CUDA？

### ISSUE THREAD

**Issue by opentld** [pytorch/pytorch#37212]
Win10，RTX2080，CUDA10.2，CUDNN7.65 ，anaconda
using conda install pytorch torchvision cudatoolkit=10.2 -c pytorch or pip torch-1.5.0-cp37-cp37m-win_amd64.whl
import torch is OK
but torch.cuda.is_available() always return false,why?

**Comment by peterjc123** [comment]
Check your driver. Is it up to date?

**Comment by WenmuZhou** [comment]
cuda 10, drive 410.104,V100 alse

**Comment by peterjc123** [comment]
@WenmuZhou Could you please share your install command and your environment by doing `python -m torch.utils.collect_env`?

**Comment by WenmuZhou** [comment]
install command is `pip3 install torch`
outptu of `python -m torch.utils.collect_env` is 
```sh
PyTorch version 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.2

Os: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: No
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0： Tesla P100-PCIE-16GB
GPU 1： Tesla P100-PCIE-16GB
GPU 2： Tesla P100-PCIE-16GB
GPU 3： Tesla P100-PCIE-16GB

Nvidia driver version: 410.104
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Version of relevant libraries:
[pip3] numpy==1.17.4
[pip3] torch==1.5.0
[pip3] torchvision==0.6.0
[conda] Could not collect
```

**Comment by opentld** [comment]
Collecting environment information...
PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Microsoft Windows 10 家庭中文版
GCC version: Could not collect
CMake version: version 3.14.0

Python version: 3.7
Is CUDA available: No
CUDA runtime version: 10.2.89
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.5.0
[pip] torchvision==0.6.0
[conda] blas                      1.0                         mkl    defaults
[conda] mkl                       2020.0                      166    defaults
[conda] mkl-include               2020.0                      166    defaults
[conda] mkl-service               2.3.0            py37hb782905_0    defaults
[conda] mkl_fft                   1.0.15           py37h14836fe_0    defaults
[conda] mkl_random                1.1.0            py37h675688f_0    defaults
[conda] torch                     1.5.0                    pypi_0    pypi
[conda] torchvision               0.6.0                    pypi_0    pypi

@peterjc123

**Comment by peterjc123** [comment]
> Collecting environment information...
> PyTorch version: 1.5.0
> Is debug build: No
> CUDA used to build PyTorch: 10.2
> 
> OS: Microsoft Windows 10 家庭中文版
> GCC version: Could not collect
> CMake version: version 3.14.0
> 
> Python version: 3.7
> Is CUDA available: No
> CUDA runtime version: 10.2.89
> GPU models and configuration: Could not collect
> Nvidia driver version: Could not collect
> cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\bin\cudnn64_7.dll
> 
> Versions of relevant libraries:
> [pip] numpy==1.18.1
> [pip] torch==1.5.0
> [pip] torchvision==0.6.0
> [conda] blas 1.0 mkl defaults
> [conda] mkl 2020.0 166 defaults
> [conda] mkl-include 2020.0 166 defaults
> [conda] mkl-service 2.3.0 py37hb782905_0 defaults
> [conda] mkl_fft 1.0.15 py37h14836fe_0 defaults
> [conda] mkl_random 1.1.0 py37h675688f_0 defaults
> [conda] torch 1.5.0 pypi_0 pypi
> [conda] torchvision 0.6.0 pypi_0 pypi
> 
> @peterjc123

What is version of your GPU driver? You can use nvidia-smi to get that info.

**Comment by opentld** [comment]
I got the problem. 
For some reasons，the video card drivers are missing，re-install drivers solved this problem.
Thank you! @peterjc123

**Comment by peterjc123** [comment]
@WenmuZhou Would you please open a new issue to get more attention?

**Comment by wcollins-ebsco** [comment]
Running on an EC2 p2 with the Deep Learning Ubuntu AMI and torch.cuda.is_available() was True. Then I pip installed torchvision, which also upgraded torch to 1.5, and now torch.cuda.is_available() is False.

**Comment by andrewjong** [comment]
I too get this issue with PyTorch 1.5 / Torchvision 0.6.0. Created a fresh conda environment, ran `conda install pytorch torchvision cudatoolkit=10.2 -c pytorch`. 

`torch.cuda.device_count()` returns `0`, `torch.cuda.is_available()` returns `False`.

However, no problems at all with PyTorch 1.4 / Torchvision 0.5.0.

**Comment by peterjc123** [comment]
Usually it is caused by the outdated GPU drivers. But however, it may also because you installed a cpu-only variant of the package.

**Comment by andrewjong** [comment]
Interesting. How come the cpu-only variant would install if I also specify cudatoolkit=10.2 in the install command? Also why would the issue occur with PyTorch 1.5 but not PyTorch 1.4, all other things being equal?

**Comment by peterjc123** [comment]
@andrewjong Sorry I don't know the answer, either. But those two are the common causes for this kind of problem. We have some smoke tests for testing the installation commands in a clean environment. But apparently, the environment is more complicated at the user's side. So they may have some unexpected results. For example, `torch.cuda.is_available()` is reported to be `False` for packages with CUDA 9.2 on Windows in my local PC. And eventually, I figured out that it is because `cudatoolkit=9.2` is not found in the defaults channel and it then picks the cpu-only variant because the feature tracking way is not working when the package of that specific version is not found. So I'd suggest you try to install them again and pay close attention to the packages that conda select for you.

**Comment by ningdez** [comment]
torch.cuda.device_count() returns **2**, torch.cuda.is_available() returns **False**.

PyTorch version:1.5.0+cu101
My cuda version is 10.1 and the nvidia driver version is 442.92. I don't know why torch.cuda.is_available() always return false? I try to change pytorch version to 1.4.0+cu101, and change nvidia driver version to 451 or 426, but the problem still exist.
Thanks a  lot.

outptu of python -m torch.utils.collect_env is

> Collecting`environment information...
> PyTorch version: 1.5.0+cu101
> Is debug build: No
> CUDA used to build PyTorch: 10.1
> 
> OS: Microsoft Windows Server 2019 Standard
> GCC version: (MinGW.org GCC Build-20200227-1) 9.2.0
> CMake version: Could not collect
> 
> Python version: 3.7
> Is CUDA available: Yes
> CUDA runtime version: 10.1.243
> GPU models and configuration:
> GPU 0: TITAN RTX
> GPU 1: TITAN RTX
> 
> Nvidia driver version: 442.92
> cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin\cudnn64_7.dll
> 
> Versions of relevant libraries:
> [pip] numpy==1.19.0
> [pip] numpydoc==0.9.1
> [pip] torch==1.5.0+cu101
> [pip] torchvision==0.6.0+cu101
> [conda] blas 1.0 mkl
> [conda] mkl 2019.4 245
> [conda] mkl-service 2.3.0 py37hb782905_0
> [conda] mkl_fft 1.0.14 py37h14836fe_0
> [conda] mkl_random 1.1.0 py37h675688f_0
> [conda] torch 1.5.0+cu101 pypi_0 pypi
> [conda] torchvision 0.6.0+cu101 pypi_0 pypi

**Comment by LiuDingbo97** [comment]
> install command is `pip3 install torch`
> outptu of `python -m torch.utils.collect_env` is
> 
> ```shell
> PyTorch version 1.5.0
> Is debug build: No
> CUDA used to build PyTorch: 10.2
> 
> Os: Ubuntu 18.04.3 LTS
> GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
> CMake version: version 3.10.2
> 
> Python version: 3.6
> Is CUDA available: No
> CUDA runtime version: 10.0.130
> GPU models and configuration:
> GPU 0： Tesla P100-PCIE-16GB
> GPU 1： Tesla P100-PCIE-16GB
> GPU 2： Tesla P100-PCIE-16GB
> GPU 3： Tesla P100-PCIE-16GB
> 
> Nvidia driver version: 410.104
> cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
> 
> Version of relevant libraries:
> [pip3] numpy==1.17.4
> [pip3] torch==1.5.0
> [pip3] torchvision==0.6.0
> [conda] Could not collect
> ```
我和你遇到了一样的问题，服务器上的cuda版本是10.0。会不会是cuda10.0不支持pytorch1.6，所以要先装一个cuda10.2。因为我每次如果在安装时指定cudatoolkit=10.0，就会自动匹配到pytorch1.4，如果我指定pytorch=1.6的话，就用不了显卡

**Comment by andrewjong** [comment]
> I too get this issue with PyTorch 1.5 / Torchvision 0.6.0. Created a fresh conda environment, ran `conda install pytorch torchvision cudatoolkit=10.2 -c pytorch`.
> 
> `torch.cuda.device_count()` returns `0`, `torch.cuda.is_available()` returns `False`.
> 
> However, no problems at all with PyTorch 1.4 / Torchvision 0.5.0.
> 
> RTX 2080 Manjaro Linux.

FYI I resolved this, 5 months later! The problem was that I had the old NVIDIA 430 drivers installed, while the newest right now is 450 or higher. @peterjc123 was correct that the problem was outdated NVIDIA drivers. I upgraded to 440 drivers (which is the highest stable on my Manjaro system right now). 

I found out however that upgrading the NVIDIA 440 drivers didn't work on my Linux 4.19 kernel - broke nvidia-smi. I upgraded my Linux kernel to 5.4 and the 440 drivers work now.

Also upgraded my system CUDA from 10.0 --> 10.2 for good measure, though probably only the driver update was necessary. 

Now the PyTorch install command on the website works perfectly. `conda install pytorch torchvision cudatoolkit=10.2 -c pytorch`. `torch.cuda.is_available()` shows `True` again, hurray!
